evennia/server/serversession.py

```diff
@@ -297,10 +294,28 @@ def msg(self, text=None, **kwargs):
         # that auto-adds the session, we'd get a kwarg collision.
         kwargs.pop("session", None)
         kwargs.pop("from_obj", None)
+
         if text is not None:
-            self.data_out(text=text, **kwargs)
-        else:
-            self.data_out(**kwargs)
+            if not (isinstance(text, str) or isinstance(text, tuple)):
+                # sanitize text before sending across the wire
+                try:
+                    text = to_str(text)
+                except Exception:
+                    text = repr(text)
+            kwargs["text"] = text
```

Comment:
What if the data comes in as, say, a list rather than a tuple? There is a reason the original `clean_senddata` is so long; it handles a lot of edge cases in order to allow for pretty generous input formats (because it's easy for devs to miss this).

Reply:
The madness of `clean_sendata` makes me want to cry. I'm not sure what to say here: I'd like to imagine that developers are reading how things work and not doing things they're not supposed to do. It's one thing to add some niceties like calling to_str or repr() on things which aren't strings, quite another to recurse arbitrary data structures to hell and back when people should have just sent the right to begin with.

I could quote the Zen of Python? Hmm.

Explicit is better than implicit.
Simple is better than complex.
Readability counts.
Special cases aren't special enough to break the rules.
Errors should never pass silently.
Unless explicitly silenced.
In the face of ambiguity, refuse the temptation to guess.

Hmmm. Maybe it would be better to start raising exceptions if bad data is fed into methods like this. But I'm trying to avoid breaking working APIs...